### ISSUE TITLE
 An argument named "subnets" is not expected here in eks-cluster.tf line 5,

### ISSUE THREAD

**Issue by mklmfane** [linuxacademy/content-terraform-2021#3]
This is the result of the  command terraform plan launched after I run terraform init -upgrade

Error: Unsupported argument
│ 
│   on eks-cluster.tf line 5, in module "eks":
│    5:   subnets         = module.vpc.private_subnets
│ 
│ An argument named "subnets" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on eks-cluster.tf line 15, in module "eks":
│   15:   workers_group_defaults = {
│ 
│ An argument named "workers_group_defaults" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on eks-cluster.tf line 19, in module "eks":
│   19:   worker_groups = [
│ 
│ An argument named "worker_groups" is not expected here.

**Comment by shakaib-arif** [comment]
I have also encountered the same issue a while back and it's related to the `eks` module version.
It can be fixed by explicitly mentioning the module version in the `eks-cluster.tf` file as following.

```
module "eks" {
  source          = "terraform-aws-modules/eks/aws"
  version         = "17.24.0"
  ...
```
I hope this might help.

**Note**: 
- I did mention to their support about this issue and hopping that they'll fix it soon. 
- There is also an [open PR ](https://github.com/linuxacademy/content-terraform-2021/pull/2) someone requested for the same issue's resolution but that involves a bit of code refactoring as well.
